src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -68,16 +64,32 @@ public static void ValidateDouble(double value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void ValidateSingle(float value)
         {
-#if BUILDING_INBOX_LIBRARY
-            if (!float.IsFinite(value))
-#else
-            if (float.IsNaN(value) || float.IsInfinity(value))
-#endif
+            if (!IsFinite(value))
             {
                 ThrowHelper.ThrowArgumentException_ValueNotSupported();
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsFinite(double value)
+        {
+#if BUILDING_INBOX_LIBRARY
+            return double.IsFinite(value);
+#else
+            return !(double.IsNaN(value) || double.IsInfinity(value));
+#endif
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
In general are we verifying that `AggressiveInlining` is necessary before we add it? Generally I would expect these to inline automatically.